pkg/supportbundle/load.go

```diff
@@ -15,28 +15,29 @@ import (
 	troubleshootclientsetscheme "github.com/replicatedhq/troubleshoot/pkg/client/troubleshootclientset/scheme"
 	"github.com/replicatedhq/troubleshoot/pkg/docrewrite"
 	"github.com/replicatedhq/troubleshoot/pkg/httputil"
+	"github.com/replicatedhq/troubleshoot/pkg/logger"
 	"github.com/replicatedhq/troubleshoot/pkg/oci"
 	"github.com/replicatedhq/troubleshoot/pkg/specs"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func GetSupportBundleFromURI(bundleURI string) (*troubleshootv1beta2.SupportBundle, error) {
+func GetSupportBundleFromURI(bundleURI string, recurse bool) (*troubleshootv1beta2.SupportBundle, error) {
 	collectorContent, err := LoadSupportBundleSpec(bundleURI)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to load collector spec")
 	}
 
 	multidocs := strings.Split(string(collectorContent), "\n---\n")
 
-	supportbundle, err := ParseSupportBundleFromDoc([]byte(multidocs[0]))
+	supportbundle, err := ParseSupportBundleFromDoc([]byte(multidocs[0]), recurse)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse collector")
 	}
 
 	return supportbundle, nil
 }
 
-func ParseSupportBundleFromDoc(doc []byte) (*troubleshootv1beta2.SupportBundle, error) {
+func ParseSupportBundleFromDoc(doc []byte, recurse bool) (*troubleshootv1beta2.SupportBundle, error) {
```

Comment:
what we could do rather than add the arg here, is to create a new function (I am totally not attached to the name):

```
func ParseSupportBundleFromDocRecursed(bundleURI string, recurse bool) (*troubleshootv1beta2.SupportBundle, error) {
   <the logic pulled from ParseSupportBundleFromDoc()>
}

```

Then have a new `ParseSupportBundleFromDoc`:

```
func ParseSupportBundleFromDoc(doc []byte) (*troubleshootv1beta2.SupportBundle, error) {
    return ParseSupportBundleFromDocRecursed(bundleURI string, false)
}
```

That way, the original function name is still there without modification (for kots import), but the new functionality is available.